system_tests/common_test.go

```diff
@@ -708,7 +717,7 @@ func getInitMessage(ctx context.Context, t *testing.T, l1client client, addresse
 }
 
 func DeployOnTestL1(
-	t *testing.T, ctx context.Context, l1info info, l1client client, chainConfig *params.ChainConfig,
+	t *testing.T, ctx context.Context, l1info info, l1client client, chainConfig *params.ChainConfig, wasmRootDir string,
```

Comment:
Deploy doesn't really care about wasmRootDir. It only cares about wasmModuleRoot (a common.Hash). In normal cases (but not in tests currently), the same wasmModuleRootDir contains multiple machines, each with it's own WasmModuleRoot.

This function could have a default of creating a newMachineLocator with path "" and reading latest from there. Another option is to not have a default always pass it a valid WasmModuleRoot.. calling test should always have a locator somewhere so it might make sense

Reply:
Done. (Although, I'm not sure it's "better".)